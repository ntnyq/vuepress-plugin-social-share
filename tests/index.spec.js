const { resolve } = require('path')
const { readFileSync } = require('fs')
const { createApp } = require('@vuepress/core')
const { pluginsWithOptions, pluginsWithoutOptions } = require('./plugins')

describe('with-options', () => {
  const app = createApp({
    sourceDir: resolve(__dirname, 'docs'),
    dest: resolve(__dirname, 'dist/with-option'),
    plugins: pluginsWithOptions,
  })

  beforeEach(async () => {
    await app.process()
    await app.build()
  }, 6e4)

  function testForFile (name, file = name) {
    test(name, () => {
      const html = readFileSync(resolve(app.outDir, file), 'utf8')

      expect(html).toMatchSnapshot()
    })
  }

  testForFile('index.html')
  testForFile('demo.html')
})

describe('without-options', () => {
  const app = createApp({
    sourceDir: resolve(__dirname, 'docs'),
    dest: resolve(__dirname, 'dist/without-option'),
    plugins: pluginsWithoutOptions,
  })

  beforeEach(async () => {
    await app.process()
    await app.build()
  }, 6e4)

  function testForFile (name, file = name) {
    test(name, () => {
      const html = readFileSync(resolve(app.outDir, file), 'utf8')

      expect(html).toMatchSnapshot()
    })
  }

  testForFile('index.html')
  testForFile('demo.html')
})
