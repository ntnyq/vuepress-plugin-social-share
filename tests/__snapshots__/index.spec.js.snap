// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic demo.md 1`] = `
<social-share />
<social-share :networks="['facebook', 'qq', 'twitter', 'line', 'reddit']" is-plain />
<social-share :networks="['twitter', 'facebook']" :tags="['SocialShare', 'VuePress']" />
<social-share :networks="['twitter', 'facebook']" />
<social-share :networks="['facebook', 'twitter']" />
<social-share :networks="['facebook', 'qq', 'twitter', 'line', 'reddit', 'skype', 'douban', 'whatsapp', 'telegram', 'weibo']" />
<social-share :networks="['email', 'pinterest', 'linkedin']" />
`;

exports[`default demo.md 1`] = `
<social-share />
<social-share :networks="['facebook', 'qq', 'twitter', 'line', 'reddit']" is-plain />
<social-share :networks="['twitter', 'facebook']" :tags="['SocialShare', 'VuePress']" />
<social-share :networks="['twitter', 'facebook']" />
<social-share :networks="['facebook', 'twitter']" />
<social-share :networks="['facebook', 'qq', 'twitter', 'line', 'reddit', 'skype', 'douban', 'whatsapp', 'telegram', 'weibo']" />
<social-share :networks="['email', 'pinterest', 'linkedin']" />
`;

exports[`no-global demo.md 1`] = `
<social-share />
<social-share :networks="['facebook', 'qq', 'twitter', 'line', 'reddit']" is-plain />
<social-share :networks="['twitter', 'facebook']" :tags="['SocialShare', 'VuePress']" />
<social-share :networks="['twitter', 'facebook']" />
<social-share :networks="['facebook', 'twitter']" />
<social-share :networks="['facebook', 'qq', 'twitter', 'line', 'reddit', 'skype', 'douban', 'whatsapp', 'telegram', 'weibo']" />
<social-share :networks="['email', 'pinterest', 'linkedin']" />
`;
